lib/web_ids.js

```diff
@@ -111,45 +95,66 @@ function checkIp(srcIp) {
 
 /**
  * Check if request body contains offending data
- * @param {Object} requestBody JSON containing request body data
+ * @param {Object} request JSON containing request data
+ * Request can have following options: request_method, request_uri,
+ * status and body
  * */
-function checkRequestBody(requestBody) {
+function checkRequest(request) {
   // TODO
   // file with signatures should be read and request uri's stored
   // match them here
-}
 
-/**
- * Check if request uri contains offending data
- * @param {String} requestUri with the uri to be accessed
- * */
-function checkRequestUri(requestUri) {
-  // TODO
-  // file with signatures should be read and request bodies stored
-  // match them here
-}
+  const method = request.request_method;
+  const uri = request.request_uri;
+  const status = request.status;
+  const body = request.body;
 
-/**
- * Check if request status contains offending data
- * @param {String} requestStatus string containing request status returned
- * */
-function checkRequestStatus(requestStatus) {
-  // TODO
-  // file with signatures should be read and request bodies stored
-  // match them here
+  const signatureMatch = false;
+
+  REQUEST_SIGNATURES.forEach((signature) => {
+    const hasMatch = false;
+
+    if (signature.method) {
+      hasMatch = method.match(signature.method);
+    }
+
+    if (signature.uri) {
+      hasMatch = uri.match(signature.ur) && hasMatch;
+    }
+
+    if (signature.status) {
+      hasMatch = status.match(signature.status) && hasMatch;
+    }
+
+    if (signature.body) {
+      hasMatch = body.match(signature.body) && hasMatch;
+    }
+
+    signatureMatch = hasMatch || signatureMatch;
+  });
+
+  if (signatureMatch) {
+    const filePath = path.join(__dirname, '..', 'logs', 'web_ids.log');
+    const type = 'Warning - potential intrusion';
+    const message = method + ' request from ip ' + request.srcip +
+                  ' tried to access ' + uri + ' and received status ' +
+                   status + ' and matched against provided signatures.';
+
+    appendToLog(filePath, type, message);
```

Comment:
Its better to set a log for each match to be able to add the signature info that resulted in the match.

The type should better be an enum that has DEBUG, INFO, WARNING, FATAL as possible values.
The message should also be shorter since it might be useful to analyse this log with other tools.

Example: method + ' "' + uri + '" from ' + request.srcip + ' returned ' + status + ' . Matched signature: ' + signature.UID

To keep the message short, a unique ID for every signature would be nice. @geo-bert  what is your take on that?

Reply:
A hard coded ID for every signature could lead to problems in extending the signatures because before adding new signatures you'd have to view all the other signature files in order to not use the same one twice. I'd rather propose a method to dynamically create the IDs with the likes of a hash function. An advantage of this could be that two of the same signatures in different files resulting in the same ID. Ideally applying the hash function twice should result in the signature again but I don't think this is feasible in our time constraints.

Reply:
but without an UID in the signature file, it is hard to reference it in the log message later.
I would set a UID in the signature file and the convention would be to have parts of the filename in the UID.
When loading the signatures, we then check if the UIDs are actually unique and if not, print out the two signatures that collide

Reply:
I agree with you but this would reduce the user experience of the user. Instead of being able to just drag and drop a signature file in, a user would need to view and possibly modify the file. For a tech savvy person this would not pose a problem but I think for less experienced people it would turn them of from using this software. If you'd be willing to make that tradeoff I am not opposed.

Reply:
Another option that would not lead to collisions would be to have IDs in a signature file such that every ID in this file is unique and in web_ids this id gets concatenated with the file name. This should lead to no collisions. This should not harm the user experience.

Reply:
I assume that signature files are only written by tech savvy people, and if the filename is part of the UID, collisions are detected by the os.
For me the benefit of an UID to reference the signature in a log outweights the possible collision problem

Reply:
Concatenation of ID and filename in web_ids is a very good idea, lets do that

Reply:
Ok. I will comment these changes to the signature files to the corresponding issue. (adding id, changing field names)

Reply:
I think it would be easier and less work to just quote the matching part in logs. My opinion is one should have enough information from the log file itself, and should not have to look into other files to find the offending signature etc. 

If the logs need to be parsed, one solution would be to use some format like YAML/TOML or something similar. 

Writing ID for each signature is cumbersome and in my opinion doesn't improve anything. 

Some log files I've seen in my career also have long lines, where each line is one entry. That's what log files are for.

Reply:
Checking how snort rules are setup for reference, they also use UIDs (called sid in their case) to reference signatures. Having a UID to reference a signature has the benefit of easily reference it when talking about a vulnerability or any other way of documenting a signature.
If you think, having the signature content also in the log message, we can do that, but then maybe just writing the part that resulted in the match. Seems cleaner for me

Reply:
Yes, but if we want custom signatures then you lose benefit of vulnerability reference since there are probably going to be many UIDs same with different file origins. If we should implement it, I would rather leave it for later, after we implement the basic functionality.

Reply:
vulnerability reference?
you are right, probably best to focus on other parts for now :D
UID implementation (or not) will be handled in the issue #9